src/reference/asciidoc/barrier.adoc

```diff
@@ -0,0 +1,87 @@
+[[barrier]]
+=== Thread Barrier
+
+Sometimes, we need to suspend a message flow thread until some other asynchronous even occurs.
+For example, consider an HTTP request that publishes a message to RabbitMQ.
+We might wish to not reply to the user until the RabbitMQ broker has issued an acknowledgment that the message was
+received.
+
+_version 4.2_ introduced the `<barrier/>` component for this purpose.
```

Comment:
I think would be better to have more full expression here for target readers:
> Since _version 4.2_ Spring Integration introduced...

Fixing on merge...